### PR TITLE
Topic/fusion 220

### PR DESCRIPTION
**Pull request by akamimoor**
## JIRA ticket
[FUSION-220](https://janrain.atlassian.net/browse/FUSION-220)

## Changes

- Created the Authenticator class to manage local and browser authentication state as well as authentication requests.
- Unit tests are added in FUSION-241

Update AppAuth-iOS SDK:
`carthage update --use-xcframeworks --platform iOS`

Confluence doc for development instructions: [Here](https://janrain.atlassian.net/l/cp/wqc2506v)

[FUSION-220]: https://janrain.atlassian.net/browse/FUSION-220?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Examples/Example-iOS_Swift-Carthage/ExampleTests/Cases/Managers/AuthenticatorTests.swift [hunk 1]
```diff
@@ -83,7 +83,7 @@ class AuthenticatorTests: XCTestCase {
             XCTAssertNotNil(discoveredConfig)
             
         } catch {
-            XCTFail("Expected to succeed while awaiting, but failed.")
+            XCTFail("Expected to sucddceed while awaiting, but failed.")
```
Comment (kreese-akamai) [inline Examples/Example-iOS_Swift-Carthage/ExampleTests/Cases/Managers/AuthenticatorTests.swift:86]: This typo snuck itself back in here 😉

**Review by kreese-akamai — COMMENTED** [review]
LGTM 👍

**Review by schaeferCode — APPROVED** [review]
Looks good!

File: Examples/Example-iOS_Swift-Carthage/ExampleTests/Cases/Managers/AuthenticatorTests.swift (hunk 1 above)
Comment (akamimoor) [reply Examples/Example-iOS_Swift-Carthage/ExampleTests/Cases/Managers/AuthenticatorTests.swift:86]: You caught me! My attempt to sneak it in did not sucddceed :(

**Review by kreese-akamai — APPROVED** [review]

**Pull request merged**